### DISCUSSION TITLE
Goal and philosophy of Helix editor

### DISCUSSION THREAD

**Discussion by Abstracting** [helix-editor/helix#10643] [General]
Hello. Freshly started user here.
Sorry if I missed it somewhere, but I'm trying to find any documentation or information on the philosophy and goal of this project.

The reason I'm curious is cause the founding goals of a project is really interesting when it comes to understanding the future of a project.

For instance, one could say that the original goal of VIM was: "Enable users to perform any kind of text manipulation tasks efficiently and is for users who require advanced features and customization in their text editing tasks"

And for Neovim i would be something like "Incorporating modern programming practices and technologies while remaining compatible with most of Vim's features and plugins."

For me, Neovim really falls down when it comes to simplicity, stability, and general out-of-the-box functionality. So my hope for Helix is that the goal is to make a parsimonious editor with minimalistic/simple yet functional and powefull features.

Is there any interview with a main contributor on my podcast, or any "state of the union" like documents? I'm always finding guiding principles interesting in projects like this, so information is appreciated!

**Comment by pascalkuthe** [comment]
https://github.com/helix-editor/helix/blob/master/docs/vision.md

**Reply by Abstracting** [reply]
Ah. Thanks. I diden't think to search for Vision. 🫣
